vistautils/io_utils.py

```diff
@@ -296,6 +342,37 @@ def is_empty(self) -> bool:
         return os.path.getsize(self._path) == 0
 
 
+@attrs(slots=True, frozen=True)
+class _FileInZipFilePathByteSource(ByteSource):
+    _zip_path = attrib_instance_of(Path)
+    _path_within_zip = attrib_instance_of(str)
+
+    def open(self) -> BytesIO:
+        # pylint:disable=not-callable
+        # pylint:disable=unused-argument
+        zip_file = ZipFile(self._zip_path, 'a')
```

Comment:
I assume you don't want to write to the zip file:
```suggestion
        zip_file = ZipFile(self._zip_path, 'r')
```